### PR TITLE
[gperftools] Add new port

### LINKED ISSUE #3536: Can vcpkg support `gperftools` and `tcmalloc` ?
rt

### PR DESCRIPTION
**Pull request by myd7349**
Fix #3536 .

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @myd7349 thanks for your pull request.

I noticed that this port failed on ` x64-uwp , arm64-windows `and `arm-uwp`. Are these expected? 

If this port doesn't currently support `uwp, arm and arm64`, please add the message to portfile.cmake.

`vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "uwp")`

**Review by NancyLi1013 — APPROVED** [review]

File: ports/gperftools/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    copy_tools(pprof pprof-symbolize)
+
+    file(REMOVE_RECURSE
+        ${CURRENT_PACKAGES_DIR}/debug/include
+        ${CURRENT_PACKAGES_DIR}/debug/share
+    )
+endif()
+
+configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (NancyLi1013) [inline ports/gperftools/portfile.cmake]: We prefer to use this to handle copyright. 
` file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`
Could you please update this?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Since this port doesn't support arm, arm64 and uwp.
Could you update [baseline ](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)with the following?
```
gperftools:arm64-windows=fail
gperftools:x64-uwp=fail
gperftools:arm-uwp=fail
```

File: ports/gperftools/CONTROL [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Source: gperftools
+Version: 2019-09-02
+Description: A set of tools for performance profiling and memory checking
+Homepage: https://github.com/gperftools/gperftools
+
+Feature: override
+Description: Override allocators
+
+Feature: tools
+Description: Build gperftools CLI tools
```
Comment (NancyLi1013) [inline ports/gperftools/CONTROL]: Could you please add `Supports: !(arm|arm64|uwp)` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/gperftools/CONTROL (hunk 2 above)
Comment (myd7349) [reply ports/gperftools/CONTROL]: Hi! @NancyLi1013 Thanks for your review. Done.

Comment (NancyLi1013) [reply ports/gperftools/CONTROL]: Thanks for your update.
It might be better to add `Supports: !(arm|arm64|uwp)` below `Homepage` filed.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
I also noticed the regressions on Linux and osx. 
Could you please try to look into them?

**Comment by NancyLi1013** [comment]
The new regressions on linux and osx:
```
There should be no bin\ directory in a static build, but /Volumes/data/work/1/s/packages/gperftools_x64-osx/bin is present.
There should be no debug\bin\ directory in a static build, but /Volumes/data/work/1/s/packages/gperftools_x64-osx/debug/bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Found 1 error(s). Please correct the portfile:
    /Volumes/data/work/1/s/ports/gperftools/portfile.cmake
```

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by NancyLi1013** [comment]
All features have passed on `x64-windows-static` triplet.
Feature `tools ` have passed on the `x86-windows` and `x64-windows` triplets.

Note: Feature `override` only supports static build.

**Comment by strega-nil** [comment]
LGTM! Thanks y'all :)

**Pull request merged**